src/index.mjs

```diff
@@ -190,13 +212,71 @@ async function processCondaEnvironment(filePath) {
     }
 }
 
+// Fetch modified lines in the PR
+async function getModifiedLines(filePath) {
+    const { context } = github;
+    const baseRef = context.payload.pull_request.base.ref;
+
+    try {
+        // Fetch the base branch to ensure we have the latest state of baseRef locally
+        await execPromise(`git fetch origin ${baseRef}`);
+
+        // Get the diff between the base branch and the current
+        const { stdout, stderr } = await execPromise(`git diff origin/${baseRef} HEAD -- ${filePath}`);
+        if (stderr) {
+            throw new Error(`Error fetching diff: ${stderr}`);
+        }
+
+        const patchLines = stdout.split('\n');
+        const modifiedLines = [];
+
+        let lineNumber = 0;
+
+        // Parse the diff to find the modified lines
+        for (const line of patchLines) {
+            if (line.startsWith('@@')) {
+                const match = /@@ -\d+,\d+ \+(\d+),/.exec(line);
+                lineNumber = match ? parseInt(match[1], 10) : lineNumber;
+            } else if (line.startsWith('+') && !line.startsWith('+++')) {
+                // Add the current line number if it's an addition
+                modifiedLines.push(lineNumber);
+                lineNumber++;
+            } else if (!line.startsWith('-')) {
+                // If it's a context line (not removed), increment the line number
+                lineNumber++;
+            }
+        }
+
+        return modifiedLines;
+    } catch (error) {
+        core.setFailed(`Error getting modified lines from commit diff: ${error.message}`);
+        return [];
+    }
+}
+
 async function run() {
     const ecosystem = core.getInput('package-ecosystem', { required: true });
+    const annotateModifiedOnly = core.getInput('annotate-modified-only') === 'true';
+    let modifiedLines = [];
 
     if (ecosystem === 'pip') {
-        await processPipRequirements('requirements.txt');
+        if (annotateModifiedOnly) {
+            modifiedLines = await getModifiedLines('requirements.txt');
+        }
+        if (modifiedLines.length > 0) {
+            await processLines('requirements.txt', modifiedLines, 'pip');
+        } else {
+            await processPipRequirements('requirements.txt');
+        }
     } else if (ecosystem === 'conda') {
-        await processCondaEnvironment('environment.yml');
+        if (annotateModifiedOnly) {
+            modifiedLines = await getModifiedLines('environment.yml');
+        }
+        if (modifiedLines.length > 0) {
+            await processLines('environment.yml', modifiedLines, 'conda');
+        } else {
+            await processCondaEnvironment('environment.yml');
+        }
```

Comment:
can you remove these if statements as per https://github.com/openteamsinc/actions/pull/8/files#r1775773852